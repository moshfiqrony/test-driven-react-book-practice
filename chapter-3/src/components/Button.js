import React from 'react';
import PropTypes from 'prop-types'

const Button = (props) => <button {...props}>{props.children}</button>

Button.propTypes={
    type: PropTypes.string
}

export default Button;