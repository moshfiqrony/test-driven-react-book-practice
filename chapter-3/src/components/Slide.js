import React from 'react';

function Slide(props) {
    return (
        <figure>
            <img alt={props.description} data-testid='image-test' src={props.imgUrl} />
            <figcaption data-testid='figcaption-test'><strong>{props.description}</strong></figcaption>
        </figure>
    );
}

export default Slide;