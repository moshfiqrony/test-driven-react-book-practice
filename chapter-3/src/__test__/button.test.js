import React from 'react';
import ReactDOM from 'react-dom';
import {render} from '@testing-library/react'
import Button from '../components/Button';
import Slide from '../components/Slide';


describe('button testing', () => {
    test('renders a button component', () => {
        const container = document.createElement('div')
        const buttonName = 'my button';
        ReactDOM.render(<Button type='button'/>, container);
        
        const button = container.querySelector('button')
        expect(button.type).toBe('button')
    })

    test('Children passed', () => {
        const container = document.createElement('div')
        const buttonName = 'my button';
        ReactDOM.render(<Button type='button'>{buttonName}</Button>, container);
        
        const button = container.querySelector('button')
        expect(button.textContent).toBe(buttonName)
    })

    test('figure rendered properly', () =>{
        const imgUrl = 'https://www.example.com/image.png'
        const description = 'this is a example image'
        const {container, getByTestId} = render(<Slide/>)
        expect(container).toMatchSnapshot();
    })

    test('image url passed using props', () =>{
        const imgUrl = 'https://www.example.com/image.png'
        const {container, getByTestId} = render(<Slide imgUrl={imgUrl} />)
        expect(getByTestId('image-test').src).toBe(imgUrl)
    })

    test('image description passed using props', () =>{
        const description = 'this is a example image'
        const {container, getByTestId} = render(<Slide description={description}/>)
        expect(getByTestId('figcaption-test').textContent).toBe(description)
    })
})


describe('stateful component testing', () => {
    test('state', () => {
        
    })
})