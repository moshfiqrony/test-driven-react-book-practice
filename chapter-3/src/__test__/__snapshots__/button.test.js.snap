// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button testing figure rendered properly 1`] = `
<div>
  <figure>
    <img
      data-testid="image-test"
    />
    <figcaption
      data-testid="figcaption-test"
    >
      <strong />
    </figcaption>
  </figure>
</div>
`;
